src/silx/gui/plot/items/core.py

```diff
@@ -655,17 +655,17 @@ def _setColormappedData(
         )
         self.__cacheColormapRange = {}  # Reset cache
 
+        colormap = self.getColormap()
         # Fill-up colormap range cache if values are provided
         if max_ is not None and numpy.isfinite(max_):
             if min_ is not None and numpy.isfinite(min_):
-                self.__cacheColormapRange[Colormap.LINEAR, Colormap.MINMAX] = min_, max_
+                self.__cacheColormapRange[Colormap.LINEAR, Colormap.MINMAX, colormap.getSaturation()] = (min_, max_)
             if minPositive is not None and numpy.isfinite(minPositive):
-                self.__cacheColormapRange[Colormap.LOGARITHM, Colormap.MINMAX] = (
+                self.__cacheColormapRange[Colormap.LOGARITHM, Colormap.MINMAX, colormap.getSaturation()] = (
```

Comment:
If it is about to cache min and max for `MINMAX` autoscale, you should not index it with `getSaturation()`. You better to put `None` or `0.0` -> Changing the saturation will not change the cached `MINMAX`.